jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/StepsRunner.java

```diff
@@ -459,13 +503,24 @@ private void pushImages() {
     results.buildResults =
         executorService.submit(
             () -> {
+              // TODO: ideally, progressDispatcher should be closed at the right moment, after the
+              // scheduled threads have completed. However, it can be tricky and cumbersome to track
+              // completion, so it may just be better to delay closing until everything ends. At
+              // least, we must ensure that it's not closed prematurely. (Garbage collection doesn't
+              // auto-close it wit the current implementation.)
+              ProgressEventDispatcher progressDispatcher =
+                  childProgressDispatcherFactory.create(
+                      "scheduling pushing manifests",
+                      results.builtImagesAndBaseImages.get().size());
+
               realizeFutures(results.applicationLayerPushResults.get());
 
               List<Future<BuildResult>> buildResults = new ArrayList<>();
               for (Map.Entry<Future<Image>, Image> entry :
                   results.builtImagesAndBaseImages.get().entrySet()) {
+                Factory progressDispatcherFactory = progressDispatcher.newChildProducer();
                 buildResults.add(
-                    pushImage(entry.getKey(), entry.getValue(), childProgressDispatcherFactory));
+                    pushImage(entry.getKey(), entry.getValue(), progressDispatcherFactory));
```

Comment:
It's ok as long as you are not accessing `progressDispatcher` concurrently. Let's keep the previous code.
```suggestion
                    pushImage(entry.getKey(), entry.getValue(), progressDispatcherFactory.newChildProducer()));
```

Reply:
I see so its only in the `executorSubmiter`